pkg/codefresh/client.go

```diff
@@ -0,0 +1,168 @@
+package codefresh
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/argoproj/argo-cd/v2/pkg/apiclient/events"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type CodefreshConfig struct {
+	BaseURL   string
+	AuthToken string
+}
+
+type CodefreshClient struct {
+	cfConfig   *CodefreshConfig
+	httpClient *http.Client
+}
+
+type CodefreshClientInterface interface {
+	Send(ctx context.Context, appName string, event *events.Event) error
+	GetApplicationConfiguration(app ApplicationIdentity) (*ApplicationConfiguration, error)
+}
+
+// VersionSource structure for the versionSource field
+type VersionSource struct {
+	File     string `json:"file"`
+	JsonPath string `json:"jsonPath"`
+}
+
+type ApplicationIdentity struct {
+	Runtime   string
+	Cluster   string
+	Namespace string
+	Name      string
+}
+
+// ApplicationConfiguration structure for GraphQL response
+type ApplicationConfiguration struct {
+	VersionSource VersionSource `json:"versionSource"`
+}
+
+// GraphQLQuery structure to form a GraphQL query
+type GraphQLQuery struct {
+	Query     string                 `json:"query"`
+	Variables map[string]interface{} `json:"variables"`
+}
+
+func NewCodefreshClient(cfConfig *CodefreshConfig) *CodefreshClient {
+	return &CodefreshClient{
+		cfConfig: cfConfig,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}
+}
+
+func (cc *CodefreshClient) Send(ctx context.Context, appName string, event *events.Event) error {
```

Comment:
Right now this client has 2 public methods.
One of them serves generic purpuse of sending any payload to the /2.0/api/events.
Second one is very specific and and submits specific graphql query to /2.0/api/graphql.
---
I propose to make this client have 2 generic methods (just rename existing) – `SendEvent` and `SendGraphQL`.
`GetApplicationConfiguration` and more other methods in the future can be implemented as part of other type `CodefreshQLInterface` or something like this:
```go
func NewCodefreshQL(cfClient *CodefreshConfig) *CodefreshQLInterface {
	return &CodefreshQLInterface{
		cfClient: cfClient
	}
}
```

Let me know WDYT.